APMrover2/RC_Channel.cpp

```diff
@@ -0,0 +1,266 @@
+
+    // @Group: 10_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[9], "10_", 10, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 11_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[10], "11_", 11, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 12_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[11], "12_", 12, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 13_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[12], "13_", 13, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 14_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[13], "14_", 14, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 15_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[14], "15_", 15, RC_Channels_Rover, RC_Channel_Rover),
+
+    // @Group: 16_
+    // @Path: RC_Channel.cpp
+    AP_SUBGROUPINFO(obj_channels[15], "16_", 16, RC_Channels_Rover, RC_Channel_Rover),
+
+    AP_GROUPEND
+};
+
+
+Mode *Rover::mode_from_mode_num(const enum Mode::number num)
+{
+    Mode *ret = nullptr;
+    switch (num) {
+    case Mode::number::MANUAL:
+        ret = &mode_manual;
+        break;
+    case Mode::number::ACRO:
+        ret = &mode_acro;
+        break;
+    case Mode::number::STEERING:
+        ret = &mode_steering;
+        break;
+    case Mode::number::HOLD:
+        ret = &mode_hold;
+        break;
+    case Mode::number::AUTO:
+        ret = &mode_auto;
+        break;
+    case Mode::number::RTL:
+        ret = &mode_rtl;
+        break;
+    case Mode::number::SMART_RTL:
+        ret = &mode_smartrtl;
+        break;
+    case Mode::number::GUIDED:
+       ret = &mode_guided;
+        break;
+    case Mode::number::INITIALISING:
+        ret = &mode_initializing;
+        break;
+    default:
+        break;
+    }
+    return ret;
+}
+
+int8_t RC_Channels_Rover::flight_mode_channel_number() const
+{
+    return rover.g.mode_channel;
+}
+
+void RC_Channel_Rover::do_mode_switch_mode_change(modeswitch_pos_t new_pos,
+                                                   modeswitch_pos_t prev_pos)
+{
+    if (new_pos < 0 || (uint8_t)new_pos > sizeof(rover.modes)) {
+        // should not have been called
+        return;
+    }
+    Mode *new_mode = rover.mode_from_mode_num((Mode::number)rover.modes[new_pos].get());
+    if (new_mode != nullptr) {
+        rover.set_mode(*new_mode, MODE_REASON_TX_COMMAND);
+    }
+}
+
+// init_aux_switch_function - initialize aux functions
+void RC_Channel_Rover::init_aux_function(const aux_func_t ch_option, const aux_switch_pos_t ch_flag)
+{
+    // init channel options
+    switch(ch_option) {
+        // at the moment nothing needs initialisation:
+        // do_aux_function(ch_option, ch_flag);
+        break;
+    default:
+        RC_Channel::init_aux_function(ch_option, ch_flag);
+        break;
+    }
+}
+
+
+bool RC_Channels_Rover::in_rc_failsafe() const
```

Comment:
I'm killing this to be a RC_Channels() concept that the vehicles don't manage at all as a part of the override stuff.